File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int cnt = 1;
            int progress = progresses[i];
            int speed = speeds[i];
            
            int lastTime = getLastTime(progress, speed);
            
            int nextCnt = 0;
            for (int j = i + 1; j < progresses.length; j++) {
                if (progresses[j] + lastTime * speeds[j] >= 100) {
                    nextCnt += 1;
                    cnt += 1;
                    i += 1;
                } else {
                    break;
                }
            }
            
            answer.add(cnt);
        }
        
        
        return changeToArray(answer);
    }
    
    private int getLastTime(int progress, int speed) {
        int lastWork = 100 - progress;
        if (lastWork % speed == 0) {
            return lastWork / speed;
        }
        return lastWork / speed + 1;
    }
    
    private int[] changeToArray(List<Integer> answer) {
        return answer.stream()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    q = deque()

    for i in range(len(progresses)):
        q.append([progresses[i], i])
    
    while q:
        
        for i in range(len(q)):
            info = q.popleft()
            progress = info[0]
            index = info[1]
            
            progress += speeds[index]
            q.append([progress, index])
            
        count = 0    
        for i in range(len(q)):
            info = q[0]
            progress = info[0]
            
            if progress >= 100:
                q.popleft()
                count += 1;
            else:
                break;
            
        
            
        if count > 0:
            answer.append(count)
        
    return answer
